File: src/insert_sort.py
def insert_sort(A):
    for i in range(1, len(A)):
        key = A[i]
        j = i - 1
        while j >= 0 and A[j] > key:
            A[j + 1] = A[j]
            j -= 1
        A[j + 1] = key


if __name__ == "__main__":
    array = [7, 2, 5, 14, 3, 8, 1, 4]
    print(f"Input array: {array}")
    insert_sort(array)
    print(f"Sorted array: {array}")

File: src/insert_sort.c
#include <stdio.h>

void insert_sort(int array[], int size){
    int i, j, key;
    for (i=1; i<size; i++){
        key = array[i];
        j = i-1;
        while (j>=0 && array[j]>key){
            array[j+1] = array[j];
            j = j-1;
        }
        array[j+1] = key;
    }
}

void print_array(int array[], int size){
    int i;
    for (i=0; i < size; i++){
        if (i == size - 1){
            printf("%d\n", array[i]);
        }
        else{
            printf("%d, ", array[i]);
        }
    }
}

int main(){
    int array[] = {7,2,5,14,3,8,1,4};
    int size = sizeof(array) / sizeof(array[0]);
    printf("Input array:\n");
    print_array(array, size);
    insert_sort(array, size);
    printf("Sorted array:\n");
    print_array(array, size);
    return 0;
}